performance/profiles/profile_matrix_cubic_spline.cc

```diff
@@ -0,0 +1,136 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "json.hpp"
+#include "json_io.hh"
+#include "profile_utility.hh"
+#include "math/interpolator.hh"
+#include "representations/calculator_spherical_expansion.hh"
+
+#include <chrono>
+
+static constexpr int N_REPETITIONS = 200;
+static unsigned int SEED = 1597463007;
+
+using namespace rascal;  // NOLINT
+
+using math::InterpolatorMatrixUniformCubicSpline;
+using math::Matrix_t;
+using math::RefinementMethod_t;
+using math::Vector_t;
+
+/* Please execute this file one time before using the profiler to create the
+ * grid.
+ */
+int main() {
+  // radial contribution parameters
+  int max_radial{3};
+  int max_angular{max_radial};
+  json fc_hypers{{"type", "Constant"},
+                 {"gaussian_sigma", {{"value", 0.5}, {"unit", "AA"}}}};
+  json hypers{
+      {"gaussian_density", fc_hypers},
+      {"max_radial", max_radial},
+      {"max_angular", max_angular},
+      {"compute_gradients", true},
+      {"cutoff_function", {{"cutoff", {{"value", 2.0}, {"unit", "AA"}}}}}};
+  auto radial_contr =
+      internal::RadialContribution<internal::RadialBasisType::GTO>(hypers);
+  std::function<Matrix_t(double)> func = [&radial_contr](double x) {
+    return radial_contr
+        .compute_contribution<internal::AtomicSmearingType::Constant>(x, 0.5);
+  };
+
+  // interpolator parameters
+  using IntpVectorUniformCubicSpline =
+      InterpolatorMatrixUniformCubicSpline<RefinementMethod_t::Exponential>;
+  std::shared_ptr<IntpVectorUniformCubicSpline> intp;
+  double x1{0};
+  double x2{8};
+  double error_bound{1e-10};
+  size_t nb_points = 1e6;
+  size_t nb_iterations = 100000;
+  const char * filename_grid{"profile_vector_cubic_spline_grid.grid"};
+  const char * filename_evaluated_grid{
+      "profile_vector_cubic_spline_grid.evaluated_grid"};
+
+  // loads grid file
+  if (not(rascal::file_exists(filename_grid)) ||
+      not(rascal::file_exists(filename_evaluated_grid))) {
+    std::cout << "Grid file does not exist, has to be computed." << std::endl;
+    Matrix_t result = func(x1);
+    int cols{static_cast<int>(result.cols())};
+    int rows{static_cast<int>(result.rows())};
+    intp = std::make_shared<IntpVectorUniformCubicSpline>(
+        func, x1, x2, error_bound, cols, rows);
+    Vector_t grid{intp->get_grid_ref()};
+    Matrix_t evaluated_grid{intp->get_evaluated_grid_ref()};
+    write_binary(filename_grid, grid);
+    write_binary(filename_evaluated_grid, evaluated_grid);
+  } else {
+    std::cout << "Grid file exists and is read." << std::endl;
+    Vector_t grid;
+    Matrix_t evaluated_grid;
+    read_binary(filename_grid, grid);
+    read_binary(filename_evaluated_grid, evaluated_grid);
```

Comment:
Is that _the_ read binary function? The one we are suspecting to be so slow?
Could not find another one.